qmk_cli/subcommands/setup.py

```diff
@@ -18,42 +19,42 @@
 @cli.argument('-y', '--yes', arg_only=True, action='store_true', help='Answer yes to all questions')
 @cli.argument('--baseurl', default='https://github.com', help='The URL all git operations start from')
 @cli.argument('-b', '--branch', default=default_branch, help='The branch to clone')
-@cli.argument('destination', default=os.environ['QMK_HOME'], nargs='?', help='The directory to clone to')
+@cli.argument('-H', '--home', default=Path(os.environ['QMK_HOME']), type=Path, help='The location for QMK Firmware. Defaults to %s' % (os.environ['QMK_HOME'],))
 @cli.argument('fork', default=default_fork, nargs='?', help='The qmk_firmware fork to clone')
 @cli.subcommand('Setup your computer for qmk_firmware.')
 def setup(cli):
-    qmk_firmware = Path(cli.args.destination)
+    """Guide the user through setting up their QMK environment.
+    """
+    clone_prompt = 'Would you like to clone {fg_cyan}%s{fg_reset} to {fg_cyan}%s{fg_reset}?' % (cli.args.fork, shlex.quote(str(cli.args.home)))
+    home_prompt = 'Would you like to set {fg_cyan}%s{fg_reset} as your QMK home?' % (shlex.quote(str(cli.args.home)),)
 
     # Sanity checks
     if cli.args.yes and cli.args.no:
         cli.log.error("Can't use both --yes and --no at the same time.")
         exit(1)
 
     # Check on qmk_firmware, and if it doesn't exist offer to check it out.
-    if (qmk_firmware / 'Makefile').exists():
-        cli.log.info('Found qmk_firmware at %s.', str(qmk_firmware))
+    if (cli.args.home / 'Makefile').exists():
+        cli.log.info('Found qmk_firmware at %s.', str(cli.args.home))
     else:
-        cli.log.error('qmk_firmware not found!')
-        if question('Would you like to clone %s?' % cli.args.fork):
+        cli.log.error('Could not find qmk_firmware!')
+        if question(clone_prompt):
             git_url = '/'.join((cli.config.setup.baseurl, cli.args.fork))
-            clone(git_url, cli.args.destination, cli.config.setup.branch)
+            result = clone(git_url, cli.args.home, cli.config.setup.branch)
+            if not result:
```

Comment:
```suggestion
            if result != 0:
```

*innocent whistling*